### ISSUE TITLE
Login doesn't work sometimes as server returns `502`

### ISSUE THREAD

**Issue by SwikritiT** [owncloud/ocis#7279]
## Describe the bug

Login doesn't work sometimes as the server returns `502`. I don't have an exact step to reproduce for this but this problem has been very prevalent in the web CI e2e tests. When the user tries to log in they are redirected to this page

![Screenshot from 2023-09-14 11-50-19](https://github.com/owncloud/ocis/assets/41103328/172b5396-3b61-4f35-b6c0-ac5f0ef46d8a)

And the tests fail.

## Network calls (web)
1. POST to `https://ocis:9200/api/v0/settings/values-list` returns 201
2. GET to `https://ocis:9200/ocs/v1.php/cloud/capabilities?format=json` returns 502
<details>
<summary> further details on the request</summary>

```
URL
https://ocis:9200/ocs/v1.php/cloud/capabilities?format=json
Request Headers
Accept: application/json, text/plain, */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en
Authorization: Bearer eyJhbGciOiJQUzI1NiIsImtpZCI6InByaXZhdGUta2V5IiwidHlwIjoiSldUIn0.eyJhdWQiOiJ3ZWIiLCJleHAiOjE2OTQ2NjUxNjMsImlhdCI6MTY5NDY2NDg2MywiaXNzIjoiaHR0cHM6Ly9vY2lzOjkyMDAiLCJqdGkiOiJPaVlFOFpjUHR0bl9WTkZBQWxwZUNCbHpiSGxzbm5EciIsImxnLmkiOnsiZG4iOiJBbGljZSBIYW5zZW4iLCJpZCI6Im93bkNsb3VkVVVJRD1kMzk3YmZhMi00ZGY0LTQ4MDYtYWJkOS05Mjg1ODQxOGEyODIiLCJ1biI6ImFsaWNlIn0sImxnLnAiOiJpZGVudGlmaWVyLWxkYXAiLCJsZy50IjoiMSIsInNjcCI6Im9wZW5pZCBwcm9maWxlIGVtYWlsIiwic3ViIjoibU9uNzFzODRrUkVMZ0d0LUBZekcxU2EzWHp2Sk1hNG1QeHlqMG9Ec1BuOW5ERXVIYXh5eVp2R3JjVlpMeVNVYVVKajJ1U2pHZHQxYkEwcHpqWGxzbWxBIn0.UQUxNnTdcjeDSg_qFplK9ImBTwYZOjpYFSiIuOm6YgtApaXiDdUI3VRI_erYmV4uLE3gWmFmJRYt1t_AFAPuhtnZ5O94DIJtf8zz19TqVTzqq47W5vQv3Th24_7Os2K3kqikJWxVjGwkmJfRGFIt0UZEfYEY9yJdurFgUaD5swHbPzfv1I1w_-Uk76eZ7coXRXi8EyHGWvYc3qlM-b2c74kHVWIZZ50aSP-USTl6py2mAufunTwHhks8-SYKPmTZ-FA35N0YULW5jNxvg6HtiOfuPNOePTHHjyzKq7bj0XomzXMAcmHnjklhQqUehl6loiv-b1v00ylT-iqu2GBN2wfMB0-dra3wVw8SZlZNoAy34x1XzELYcwAC5FPqL0oai4mE-xLkdyZIbujgyYBj17i2eeqCIDP7SDjrGFAKJdm7VQh84Ya8YGWMIat89BDbEcaySEFFrCCc18E-DuQmwoY9axz89DjX3PiIdOSPR61SV_Haz4lvmeWArUSskqKqvfv63s8OWfaBkff8rlBeO0kWVy-KoSUHILxICiRkXX_gf6y1H3UuCUAi0nA3i8YpFT9AlviVVS81wT4-gRNYnjr91ddVyJA3xsOXmcXM6qhzfrKA38oghnTK4qT1iJ94W5pVOluYCknbBfdDvZ9mHMIZyrHez-A3L6vNF8nYXvs
Connection: keep-alive
Host: ocis:9200
Referer: https://ocis:9200/oidc-callback?code=gXNZeoxUwcuYMGXBx2qZSBA6GJaW5vTP&scope=email%20openid%20profile&session_state=bc840386e646625206a5661165e93640035ae209499816f9c9edb9c56ae826b8.OvPNfg9AiiU0gLNJEWd3Y7wn_wBgLetjHT_BTQMPtL4&state=5c9c5a5e46364c0ea50d891756cfaa33
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/114.0.5735.198 Safari/537.36
X-Request-ID: 720c7a8a-4154-430a-aaab-07f7ee10a1d8
X-Requested-With: XMLHttpRequest
sec-ch-ua: "Not.A/Brand";v="8", "Chromium";v="114", "HeadlessChrome";v="114"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Linux"
Response Headers
Content-Length: 0
Date: Thu, 14 Sep 2023 04:14:23 GMT
X-Request-Id: 720c7a8a-4154-430a-aaab-07f7ee10a1d8
Response Body
```
</details>

3. GET to `https://ocis:9200/ocs/v1.php/cloud/user` also returns 502 

<details>
<summary> further details on the request</summary>

```
URL
https://ocis:9200/ocs/v1.php/cloud/user
Request Headers
Accept: */*
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Host: ocis:9200
Referer: https://ocis:9200/oidc-callback?code=gXNZeoxUwcuYMGXBx2qZSBA6GJaW5vTP&scope=email%20openid%20profile&session_state=bc840386e646625206a5661165e93640035ae209499816f9c9edb9c56ae826b8.OvPNfg9AiiU0gLNJEWd3Y7wn_wBgLetjHT_BTQMPtL4&state=5c9c5a5e46364c0ea50d891756cfaa33
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/114.0.5735.198 Safari/537.36
accept-language: en
authorization: Bearer eyJhbGciOiJQUzI1NiIsImtpZCI6InByaXZhdGUta2V5IiwidHlwIjoiSldUIn0.eyJhdWQiOiJ3ZWIiLCJleHAiOjE2OTQ2NjUxNjMsImlhdCI6MTY5NDY2NDg2MywiaXNzIjoiaHR0cHM6Ly9vY2lzOjkyMDAiLCJqdGkiOiJPaVlFOFpjUHR0bl9WTkZBQWxwZUNCbHpiSGxzbm5EciIsImxnLmkiOnsiZG4iOiJBbGljZSBIYW5zZW4iLCJpZCI6Im93bkNsb3VkVVVJRD1kMzk3YmZhMi00ZGY0LTQ4MDYtYWJkOS05Mjg1ODQxOGEyODIiLCJ1biI6ImFsaWNlIn0sImxnLnAiOiJpZGVudGlmaWVyLWxkYXAiLCJsZy50IjoiMSIsInNjcCI6Im9wZW5pZCBwcm9maWxlIGVtYWlsIiwic3ViIjoibU9uNzFzODRrUkVMZ0d0LUBZekcxU2EzWHp2Sk1hNG1QeHlqMG9Ec1BuOW5ERXVIYXh5eVp2R3JjVlpMeVNVYVVKajJ1U2pHZHQxYkEwcHpqWGxzbWxBIn0.UQUxNnTdcjeDSg_qFplK9ImBTwYZOjpYFSiIuOm6YgtApaXiDdUI3VRI_erYmV4uLE3gWmFmJRYt1t_AFAPuhtnZ5O94DIJtf8zz19TqVTzqq47W5vQv3Th24_7Os2K3kqikJWxVjGwkmJfRGFIt0UZEfYEY9yJdurFgUaD5swHbPzfv1I1w_-Uk76eZ7coXRXi8EyHGWvYc3qlM-b2c74kHVWIZZ50aSP-USTl6py2mAufunTwHhks8-SYKPmTZ-FA35N0YULW5jNxvg6HtiOfuPNOePTHHjyzKq7bj0XomzXMAcmHnjklhQqUehl6loiv-b1v00ylT-iqu2GBN2wfMB0-dra3wVw8SZlZNoAy34x1XzELYcwAC5FPqL0oai4mE-xLkdyZIbujgyYBj17i2eeqCIDP7SDjrGFAKJdm7VQh84Ya8YGWMIat89BDbEcaySEFFrCCc18E-DuQmwoY9axz89DjX3PiIdOSPR61SV_Haz4lvmeWArUSskqKqvfv63s8OWfaBkff8rlBeO0kWVy-KoSUHILxICiRkXX_gf6y1H3UuCUAi0nA3i8YpFT9AlviVVS81wT4-gRNYnjr91ddVyJA3xsOXmcXM6qhzfrKA38oghnTK4qT1iJ94W5pVOluYCknbBfdDvZ9mHMIZyrHez-A3L6vNF8nYXvs
content-type: application/x-www-form-urlencoded
ocs-apirequest: true
sec-ch-ua: "Not.A/Brand";v="8", "Chromium";v="114", "HeadlessChrome";v="114"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Linux"
x-request-id: c05a0ec0-1027-497b-a62f-c902424119fe
x-requested-with: XMLHttpRequest
Response Headers
Content-Length: 0
Date: Thu, 14 Sep 2023 04:14:23 GMT
X-Request-Id: c05a0ec0-1027-497b-a62f-c902424119fe
Response Body
```
</details>

## Server logs
```console
{"level":"error","service":"frontend","error":"com.owncloud.api.gateway: service not found","time":"2023-09-12T00:35:03.519061702Z","message":"cannot get gateway client"} {"level":"error","service":"frontend","error":"com.owncloud.api.gateway: service not found","server":"frontend","time":"2023-09-12T00:35:03.519325126Z","message":"Shutting down server"}
```

I couldn't reproduce this error locally but we have been seeing this a lot in the CI of the web 
Build of test failure - https://drone.owncloud.com/owncloud/web/38735/12/19

**Comment by SwikritiT** [comment]
related test failure in web https://github.com/owncloud/web/issues/9687

**Comment by kobergj** [comment]
gateway service did not start somehow. No clue why yet. Most requests go through gateway, thats why we get `502` everywhere. Requests to `setting` service go directly to settings thats why it works. 

Not really sure why gateway is not booting up. But it doesn't crash, it never started. Can you report if this happens again? Maybe we can extract the issue when we compare multiple failing pipelines.

**Comment by SwikritiT** [comment]
@kobergj It happens a lot on the web. We have been seeing this for quite a while now.
These are some of the build links that I could find
https://drone.owncloud.com/owncloud/web/38732/11/19
https://drone.owncloud.com/owncloud/web/38708/12/19
https://drone.owncloud.com/owncloud/web/38707/12/19
https://drone.owncloud.com/owncloud/web/38669/11/19

This has only happened in the `e2e-tests-oCIS-app-provider` pipeline in the CI but @SagarGi was able to reproduce this on his local machine, it hasn't been reproducible for me yet.

**Comment by saw-jan** [comment]
Seen `502` error in webUI acceptance tests: https://drone.owncloud.com/owncloud/ocis/26531/87/13

```feature
✔ And user "David" has been created with default attributes and without skeleton files in the server # stepDefinitions/middlewareContext.js:66
✖ And user "Alice" has uploaded file "testavatar.png" to "simple-folder/testimage.png" in the server # stepDefinitions/middlewareContext.js:66
    Error: Error: HTTP Request Failed: Could not create the file "simple-folder/testimage.png" for user "Alice"., Status: 502 undefined
```

**Comment by saw-jan** [comment]
Seen again in api tests: https://drone.owncloud.com/owncloud/ocis/26536/66/5
```bash
Failed step: And user "Alice" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt"
HTTP status code was not 201 or 204 while trying to upload file '/textfile0.txt' for user 'Alice'
Failed asserting that an array contains 502.
```

**Comment by kobergj** [comment]
@SagarGi can you tell me how to reproduce this locally? Seems the registry is not working correctly which is strange because it is supposed to be inmemory.

**Comment by SwikritiT** [comment]
Now it's reproducible for me too, in master of web I ran all the services from docker
```
docker compose up traefik ocis wopiserver ocis-appprovider-onlyoffice onlyoffice ocis-appprovider-collabora collabora

```
Now I'm in this loop

https://github.com/owncloud/ocis/assets/41103328/b7d0bfb1-bc96-4daf-b842-efd56f0c68e2

**Comment by saw-jan** [comment]
Here is the full log of ocis when it occurs:
<details>
<summary>ocis service log</summary>

```bash
web_ocis  | 
web_ocis  | =========================================
web_ocis  |  generated OCIS Config
web_ocis  | =========================================
web_ocis  |  configpath : /etc/ocis/ocis.yaml
web_ocis  |  user       : admin
web_ocis  |  password   : admin
web_ocis  | 
web_ocis  | {"level":"info","time":"2023-09-15T05:18:33.830470935Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0001_create_spaces_directory_structure.go:40","message":"Migrating spaces directory structure..."}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:33.830555745Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0002_move_spacetypes_to_indexes.go:38","message":"Migrating space types indexes..."}
web_ocis  | {"level":"info","root":"/var/lib/ocis/storage/metadata","time":"2023-09-15T05:18:33.831059763Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0003_switch_to_messagepack_metadata.go:52","message":"Migrating to messagepack metadata backend..."}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:33.831092999Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0003_switch_to_messagepack_metadata.go:113","message":"done."}
web_ocis  | {"level":"warn","error":"open /var/lib/ocis/storage/metadata/indexes/by-user-id: no such file or directory","time":"2023-09-15T05:18:33.858408978Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:42","message":"error listing user indexes"}
web_ocis  | {"level":"warn","error":"open /var/lib/ocis/storage/metadata/indexes/by-group-id: no such file or directory","time":"2023-09-15T05:18:33.858431419Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:62","message":"error listing group indexes"}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:33.858449406Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:96","message":"done."}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:33.861514649Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0005_fix_messagepack_space_index_format.go:72","message":"done."}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:34.693325841Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0001_create_spaces_directory_structure.go:40","message":"Migrating spaces directory structure..."}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:34.693372447Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0002_move_spacetypes_to_indexes.go:38","message":"Migrating space types indexes..."}
web_ocis  | {"level":"info","root":"/var/lib/ocis/storage/users","time":"2023-09-15T05:18:34.693485118Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0003_switch_to_messagepack_metadata.go:52","message":"Migrating to messagepack metadata backend..."}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:34.693507121Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0003_switch_to_messagepack_metadata.go:113","message":"done."}
web_ocis  | {"level":"warn","error":"open /var/lib/ocis/storage/users/indexes/by-user-id: no such file or directory","time":"2023-09-15T05:18:34.695255478Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:42","message":"error listing user indexes"}
web_ocis  | {"level":"warn","error":"open /var/lib/ocis/storage/users/indexes/by-group-id: no such file or directory","time":"2023-09-15T05:18:34.695268031Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:62","message":"error listing group indexes"}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:34.695280229Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:96","message":"done."}
web_ocis  | {"level":"info","time":"2023-09-15T05:18:34.728173513Z","caller":"github.com/cs3org/reva/v2@v2.16.1-0.20230911153145-a2e2320f3448/pkg/storage/utils/decomposedfs/migrator/0005_fix_messagepack_space_index_format.go:72","message":"done."}
web_ocis  | 2023-09-15T05:20:26Z ERR failed to authenticate the request error="failed to verify access token: crypto/rsa: verification error" authenticator=oidc path=/api/v0/settings/values-list service=proxy
web_ocis  | 2023-09-15T05:20:26Z ERR failed to authenticate the request error="failed to verify access token: crypto/rsa: verification error" authenticator=oidc path=/ocs/v1.php/cloud/user service=proxy
web_ocis  | 2023-09-15T05:20:27Z ERR error getting auth provider client error="internal error: gateway: error finding an auth provider for type: bearer" pkg=rgrpc service=gateway traceid=00000000000000000000000000000000 type=bearer
web_ocis  | 2023/09/15 05:20:32 http: proxy error: context canceled
web_ocis  | 2023/09/15 05:20:32 http: proxy error: context canceled
web_ocis  | 2023-09-15T05:20:36Z ERR error getting auth provider client error="internal error: gateway: error finding an auth provider for type: bearer" pkg=rgrpc service=gateway traceid=00000000000000000000000000000000 type=bearer
web_ocis  | 2023-09-15T05:21:12Z ERR failed to list storage spaces error="error reading user index: stat /var/lib/ocis/storage/users/indexes/by-user-id/cc62ce6b-6e74-4575-b251-726fbae24a08.mpk: no such file or directory" filters=[{"Term":{"SpaceType":"personal"},"type":4},{"Term":{"User":{"opaque_id":"cc62ce6b-6e74-4575-b251-726fbae24a08"}},"type":6}] pkg=rgrpc service=storage-users status={"code":15,"message":"error listing spaces","trace":"00000000000000000000000000000000"} traceid=00000000000000000000000000000000
web_ocis  | 2023/09/15 05:21:56 Could not create config: config in /etc/ocis/ocis.yaml already exists
web_ocis  | 2023-09-15T05:23:10Z ERR error getting auth provider client error="internal error: gateway: error finding an auth provider for type: bearer" pkg=rgrpc service=gateway traceid=00000000000000000000000000000000 type=bearer
web_ocis  | 2023/09/15 05:23:21 http: proxy error: context canceled
web_ocis  | 2023/09/15 05:23:21 http: proxy error: context canceled
web_ocis  | 2023-09-15T05:23:23Z ERR error getting auth provider client error="internal error: gateway: error finding an auth provider for type: bearer" pkg=rgrpc service=gateway traceid=00000000000000000000000000000000 type=bearer
web_ocis  | 2023-09-15T05:23:29Z ERR Shutting down server service=invitations transport=http
web_ocis  | 2023-09-15T05:23:29Z ERR Shutting down server server=server service=ocdav
web_ocis  | 2023-09-15T05:23:29Z ERR Failed to register prometheus metric error="duplicate metrics collector registration attempted" metric=BuildInfo service=invitations
web_ocis  | 2023-09-15T05:23:29Z ERR Failed to register prometheus metric error="duplicate metrics collector registration attempted" metric=counter service=invitations
web_ocis  | 2023/09/15 05:23:57 Could not create config: config in /etc/ocis/ocis.yaml already exists
web_ocis  | 2023-09-15T05:23:57Z ERR cannot get gateway client error="com.owncloud.api.gateway: service not found" service=frontend
web_ocis  | 2023-09-15T05:23:57Z ERR Shutting down server error="com.owncloud.api.gateway: service not found" server=frontend service=frontend
web_ocis  | 2023-09-15T05:24:03Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:03 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:03Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:03 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:08Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:08 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:08Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:08 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:13Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:13 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:13Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:13 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:18Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:18 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:18Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:18 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023/09/15 05:24:24 http: proxy error: context canceled
web_ocis  | 2023/09/15 05:24:24 http: proxy error: context canceled
web_ocis  | 2023-09-15T05:24:24Z ERR IdentifierIdentityManager: fetch failed to get user from userID error="ldap identifier backend get user connect error: LDAP Result Code 200 \"Network Error\": dial tcp 127.0.0.1:9235: operation was canceled" service=idp
web_ocis  | 2023-09-15T05:24:26Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:26 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:26Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:26 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:30Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:30 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:30Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:30 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:39Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:39 http: proxy error: unsupported protocol scheme ""
web_ocis  | 2023-09-15T05:24:39Z ERR could not select service from the registry error="not found" service=com.owncloud.web.frontend
web_ocis  | 2023/09/15 05:24:39 http: proxy error: unsupported protocol scheme ""
```
</details>

#### Possible Steps to Reproduce:
1. Start ocis (I used web docker-compose)
    ```bash
    docker compose up traefik ocis wopiserver ocis-appprovider-onlyoffice onlyoffice ocis-appprovider-collabora collabora
    ```
2. Try to access ocis **immediately** from https://host.docker.internal:9200
3. Cannot login (loop in "Not logged in" page)

**Comment by SwikritiT** [comment]
The `502` error is now occurring very frequently in ap tests as well https://drone.owncloud.com/owncloud/ocis/26552/66/5, it's very difficult to get a green CI.

**Comment by saw-jan** [comment]
API tests mostly fail with upload resource step
```js
{
  "level": "error",
  "service": "com.owncloud.web.frontend",
  "error": "not found",
  "time": "2023-09-15T03:51:51.21133982Z",
  "message": "could not select service from the registry"
}
2023/09/15 03:51:51 http: proxy error: unsupported protocol scheme ""
{
  "level": "error",
  "service": "ocdav",
  "name": "com.owncloud.web.ocdav",
  "traceid": "00000000000000000000000000000000",
  "request-id": "coreApiSharePublicLink1/accessToPublicLinkShare.feature:12-13",
  "path": "/users/fea29ff4-3925-4442-aab4-66ca804af5d1/testavatar.jpg",
  "time": "2023-09-15T03:51:51.211757118Z",
  "message": "PUT request to data server failed"
}
```

**Comment by micbar** [comment]
This is becoming a blocker because it affects the CI.

**Comment by micbar** [comment]
Closed by #6751
